### ISSUE TITLE
New files on branches not reflected

### ISSUE THREAD

**Issue by slafs** [codecov/support#59]
I made a small repo (using Python coverage) to visualize my issue https://github.com/slafs/codecov-issue.

If I introduce a new file in a commit that's on a different branch (than a default one) then the codecov report doesn't seem to see this file. Check out [this report of feature/more-tests branch](https://codecov.io/github/slafs/codecov-issue/my_package?ref=86cb0864f6e9ab08e5b998ac45b45753a30062b7). This report refers to slafs/codecov-issue@86cb0864f6 which introduced a new test file `my_package/tests/test_my_other_function.py` (this also changed coverage from 80% to 100%) but this file is not visible on the codecov report.

When I add a new file on default branch (`master` in this case) the report seems to be OK - see https://codecov.io/github/slafs/codecov-issue/my_package/tests?ref=fb38384bf27d5301a2223b60d083ca2f41368262

**Comment by stevepeak** [comment]
Thanks for the thorough report! :+1: 

Can you try this again but expose the coverage report via

```diff
 commands =
     coverage run --source my_package {envbindir}/py.test
     coverage report
+   coverage xml && cat coverage.xml
     codecov
```

I'm curious if the xml report is missing the new test file.

**Comment by slafs** [comment]
On coverage xml report the file seems to be present. See https://gist.github.com/slafs/47fc8daa33eb4ecdfda3

**Comment by stevepeak** [comment]
This is indeed odd... I'm not sure what to make of it. I take that same report in your provided Gist and it comes out the following files covered:

```
my_package/tests/__init__.py
my_package/tests/test_my_other_function.py
my_package/tests/test_my_function.py
my_package/__init__.py
my_package/my_module.py
```

Can you recreate this issue? Before doing it I'd like to enable a debug mode on my backend. We can chat in https://gitter.im/codecov/support about it more. Thanks!

**Comment by slafs** [comment]
Oh sorry, I just run `coverage xml` and put it into that gist. Actually when I do an explicit `coverage xml` before doing `codecov`. The report seems to be fine https://codecov.io/github/slafs/codecov-issue/my_package/tests?ref=17d0f8fb990aab3bfa1a3253656dfcc36c470aea

Ok... weird... by the time I managed to write the above sentence I tried also to reproduce the issue on a different branch without the `coverage xml` part and... it works now :confused:. See for yourself https://codecov.io/github/slafs/codecov-issue/my_package/tests?ref=e4e820e66cfd4f32cdfbe2ee664003bcae9eac33

But this bug is somewhere out there, since my company's codecov reports are also missing files.
